apps/prairielearn/src/pages/clientFilesCourse/clientFilesCourse.js

```diff
@@ -1,28 +1,41 @@
+// @ts-check
+import { selectCourseById } from '../../models/course';
+import { selectQuestionById } from '../../models/question';
+
+const asyncHandler = require('express-async-handler');
 const path = require('path');
 const express = require('express');
-const router = express.Router();
 
 const error = require('@prairielearn/error');
 const chunks = require('../../lib/chunks');
-const ERR = require('async-stacktrace');
 
-router.get('/*', function (req, res, next) {
-  const filename = req.params[0];
-  if (!filename) {
-    return next(
-      error.make(400, 'No filename provided within clientFilesCourse directory', {
-        locals: res.locals,
-        body: req.body,
-      }),
-    );
-  }
-  const coursePath = chunks.getRuntimeDirectoryForCourse(res.locals.course);
-  chunks.ensureChunksForCourse(res.locals.course.id, { type: 'clientFilesCourse' }, (err) => {
-    if (ERR(err, next)) return;
+module.exports = function (options = { publicEndpoint: false }) {
+  const router = express.Router({ mergeParams: true });
+  router.get(
+    '/*',
+    asyncHandler(async function (req, res) {
+      if (options.publicEndpoint) {
+        res.locals.course = await selectCourseById(req.params.course_id);
+        res.locals.question = await selectQuestionById(req.params.question_id);
```

Comment:
oh no! ya I think you are right. I will add that validation. 

As far as making it clear, this is already in the documentation for Question Sharing. Is there somewhere else you would like to have it repeated as well?

Reply:
oh sorry, it is in the docs about sharing questions generally, but it doesn't specifically talk about public sharing (speaking of which, I need to add something about public sharing to the docs anyway...)

Reply:
Yeah, I think it's important to discuss public sharing specifically since sharing to the world inherently requires a little more caution.

I think `publicQuestionPreview` needs the same check to ensure that the course ID matches the question's course ID.

Reply:
I've added that check to publicQuestionPreview, and added new documentation for Public sharing!

Reply:
uggh I just remembered we actually did not implement `clientFilesCourse` for shared questions before this. But in this PR I implemented it for the public previews. That's kind of a mess. 

I think for consistency, I am just going to remove the support for `clientFilesCourse` from this PR, and add it back in once we are ready to implement (doing so is going to be somewhat complicated: https://github.com/PrairieLearn/PrairieLearn/issues/8322). Does that sound all right?

Reply:
I think that's reasonable, yeah. Great catch!

Reply:
All done, ready to go! The changes to the `clientFilesCourse.js` file are now purely cosmetic (moved to async).